File: Programmers/KAKAO/k진수에서 소수 개수 구하기.cpp
// 2022 KAKAO BLIND RECRUITMENT level 2
#define LL long long
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

bool is_prime(LL number) {
    if (number == 0 || number == 1)
        return false;
    
    for (LL i = 2; i <= (LL)sqrt(number); ++i) {
        if (number % i == 0) 
            return false;
    }
    
    return true;
}

string trans_number(int n, int k) {
    string ret = "";
    
    while(n > 0) {
        ret += to_string(n % k);
        n /= k;
    }

    return string(ret.rbegin(), ret.rend());;
}

int solution(int n, int k) {
    int answer = 0;
    
    string t_n = trans_number(n, k);
    LL size = t_n.length();
    string prime = "";
    
    for (LL i = 0; i < size; ++i) {
        // 소수 체크
        if (t_n[i] == '0' && prime.length() > 0) {
            if (is_prime(stoll(prime)))
                answer++;
            prime = "";
        }
        
        else {
            prime += t_n[i];
        }
    }
    
    if (prime.length() > 0) {
        if (is_prime(stoll(prime)))
            answer++;
    }
    
    return answer;
}

File: Programmers/KAKAO/k진수에서 소수 개수 구하기.py
# 2022 KAKAO BLIND RECRUITMENT level 2
import math

# 진법 변환
def trans_Number(n, k):
    ret = ""
    
    while n > 0:
        ret += str(n%k)
        n //= k
    
    return reversed(ret)

# 소수 판별
def is_Prime(number):
    if number == 0 or number == 1:
        return False
    for i in range(2, int(math.sqrt(number)) + 1):
        if number % i == 0:
            return False
    return True

def solution(n, k):
    answer = 0
    ret = trans_Number(n, k) # 변환
    prime_candi = ""
    
    for r in ret:
        # P0 0P0 ,0P 에서 중복되는 소수를 발견하더라도 따로 세야함
        if r == "0":
            if len(prime_candi) and is_Prime(int(prime_candi)):
                answer += 1
            prime_candi = ""
        else:
            prime_candi += r
    
    # 만약 소수 후보가 존재할 경우 체크
    if len(prime_candi) and is_Prime(int(prime_candi)):
        answer += 1
    return answer